core/src/toga/statusicons.py

```diff
@@ -0,0 +1,247 @@
+from __future__ import annotations
+
+import sys
+from collections.abc import Iterator
+from typing import TYPE_CHECKING, Mapping, Sequence
+
+import toga
+from toga.command import Command, CommandSet, Group
+from toga.handlers import wrapped_handler
+from toga.icons import Icon
+from toga.platform import get_platform_factory
+
+if TYPE_CHECKING:
+    from toga.icons import IconContentT
+
+_py_id = id
+
+
+class BaseStatusIcon:
+    def __init__(self, icon: IconContentT | None = None, **kwargs):
+        """An abstract base class for all status icons."""
+        super().__init__(**kwargs)
```

Comment:
In isolation it's not obvious why a top-level class would call `super`. It would be clearer for `MenuStatusIcon` to call `__init__` on both of its base classes directly, without using `super`. Then these `**kwargs` can be removed.

Reply:
Yeah - MRO is a bit of a pain with complex hierarchies. I went with the "least specific" implementation, but I agree that explicitly calling the constructors you want is more obvious on first reading.